### ISSUE TITLE
Update to latest wp-scripts

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/oik-magnetic-poetry#6]
oik-magnetic-poetry is currently built with @wordpress/scripts v15.0.1
Latest version is 26.15.0
It needs updating and rebuilding.
`npm run packages-update`

**Comment by bobbingwide** [comment]
Delivered in v0.3.0

**Comment by bobbingwide** [comment]
latest wp-scripts is now v30.0.2.
I've updated, but the build fails instantly. Bugger

[webpack-cli] TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at getNormalizedEntryStatic (C:\apache\htdocs\wordpress\wp-content\plugins\oik-magnetic-poetry\node_modules\webpack\lib\config\normalization.js:491:27)

**Comment by bobbingwide** [comment]
> but the build fails instantly.

Resolved by reverting to v29.0.0 and adding `webpack.config.js`.
I assume that I can now update to the latest version. 

I also need to check that my internationalisation logic still works / is necessary.

**Comment by bobbingwide** [comment]
I also needed to add clsx to replace classnames. See 
https://github.com/bobbingwide/bobbingwide/issues/122

**Comment by bobbingwide** [comment]
> I also need to check that my internationalisation logic still works / is necessary.

In the filter function `oikmp_load_script_textdomain_relative_path()`
the original effective code was 

```
if ( false !== strrpos( $relative, './build/index.js' )) {
   $relative = 'build/index.js';
}
```
I copied and cobbled the code from oik-css to produce this
```
bw_trace2();
if ( false !== strpos( $src, '/oik-magnetic-poetry/src/' )) {
	$relative='build/' . basename( $relative );
	bw_trace2( $relative, "relative update");
}
```

But when we look at the trace output from the first trace call it would appear that both the `$relative` and `$src` params are already OK.  

C:\apache\htdocs\wordpress\wp-content\plugins\oik-magnetic-poetry\oik-magnetic-poetry.php(70:0) oikmp_load_script_textdomain_relative_path(12) 131 0 2024-09-23T10:37:35+00:00 4.286424 0.001568 cf=admin_print_scripts,load_script_textdomain_relative_path 43309 79 1934<>4157 67108864/69206016 256M F=1056 2 Array

    [0] => (string) "build/magnetic-poetry.js"
    [1] => (string) "https://dev.s.b/cwiccer/wp-content/plugins/oik-magnetic-poetry/build/magnetic-poetry.js"

This suggests I don't need the filter function for `load_script_textdomain_relative_path` any more.

**Comment by bobbingwide** [comment]
When building with 29.0.0 I get these message from `npm run build`.
```
built modules 3.49 KiB (javascript) 1.02 KiB (css/mini-extract) [built]
  ./src/oik-magnetic-poetry/index.js + 9 modules 3.49 KiB [not cacheable] [built] [code generated]
  css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[3].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[3].use[2]!./node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[3].use[3]!./src/oik-magnetic-poetry/style.scss 966 bytes [built] [code generated]
  css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[3].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[3].use[2]!./node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[3].use[3]!./src/oik-magnetic-poetry/editor.scss 77 bytes [built] [code generated]

LOG from ./node_modules/sass-loader/dist/cjs.js sass-loader ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[3].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[3].use[2]!./node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[3].use[3]!./src/oik-magnetic-poetry/style.scss
<w> Deprecation The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.
<w>
<w> More info: https://sass-lang.com/d/legacy-js-api
<w>
<w> null

LOG from ./node_modules/sass-loader/dist/cjs.js sass-loader ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[3].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[3].use[2]!./node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[3].use[3]!./src/oik-magnetic-poetry/editor.scss
<w> Deprecation The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.
<w>
<w> More info: https://sass-lang.com/d/legacy-js-api
<w>
<w> null

webpack 5.94.0 compiled successfully in 1141 ms
```
**Do I need to be worried about them? Or will they go away with v30.0.2?**

**Comment by bobbingwide** [comment]
> Or will they go away with v30.0.2

Answer: No.

**Comment by bobbingwide** [comment]
> Resolved by reverting to v29.0.0 and adding webpack.config.js.

This wasn't the first time I had this problem. See 
https://github.com/bobbingwide/oik-css/issues/26#issuecomment-1994001428

**Comment by bobbingwide** [comment]
Delivered in v0.4.0 on oik-plugins.co.uk and oik-plugins.com
